runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/HttpEngineEventListener.kt

```diff
@@ -2,30 +2,31 @@
  * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0.
  */
-package aws.smithy.kotlin.runtime.http.engine
+package aws.smithy.kotlin.runtime.http.engine.okhttp
 
+import aws.smithy.kotlin.runtime.http.operation.HttpOperationContext
 import aws.smithy.kotlin.runtime.logging.Logger
+import aws.smithy.kotlin.runtime.util.get
 import okhttp3.*
 import java.io.IOException
 import java.net.InetAddress
 import java.net.InetSocketAddress
 import java.net.Proxy
 
-// FIXME - propagate call metadata (e.g. sdkRequestId) using Request.tag instead of abusing AMZ_SDK_INVOCATION_ID. Requires more direct integration with okhttp though.
-private const val AMZ_SDK_INVOCATION_ID_HEADER = "amz-sdk-invocation-id"
-
 internal class HttpEngineEventListener(
     private val pool: ConnectionPool
 ) : EventListener() {
     private val logger = Logger.getLogger<HttpEngineEventListener>()
 
     private inline fun traceCall(call: Call, crossinline msg: () -> Any) {
-        val sdkRequestId = call.request().header(AMZ_SDK_INVOCATION_ID_HEADER)
+        val sdkRequestTag = call.request().tag<SdkRequestTag>()
+        val sdkRequestId = sdkRequestTag?.execContext?.getOrNull(HttpOperationContext.SdkRequestId)
```

Comment:
**Question**: What does it mean if any of these are null and we cannot derive the `sdkRequestId`?

Reply:
As in what will happen? Nothing, the log statement will have `[sdkRequestId=null]`. In practice this should never happen in an actual generated SDK because we guarantee that element to be present at the time the request is constructed. The reason I've relaxed it to `getOrNull` is that many of the unit tests here and in the integration test suite don't include this additional `ExecutionContext` setup.